src/main/scala/is/hail/expr/ir/Interpret.scala

```diff
@@ -183,6 +154,17 @@ object Interpret {
               }
             }
         }
+      case ApplyComparisonOp(op, l, r) =>
+        val lValue = interpret(l, env, args, agg)
+        val rValue = interpret(r, env, args, agg)
+        op match {
+          case EQ(_) => lValue == rValue
```

Comment:
do these treat nan / infinity correctly?

Reply:
```
scala> Float.NaN == Float.NaN
res0: Boolean = false

scala> Float.PositiveInfinity == Float.PositiveInfinity
res1: Boolean = true
```

Reply:
yeah. CodeOrdering should be implementing this definition of equality.